bin/od

```diff
@@ -63,15 +62,26 @@ getopts('A:bcdfij:lovx') or help();
     defined $opt_A ? ($radix = $opt_A) : ($radix = 'o');
     defined $opt_j && ($offset1 = $opt_j);
 
-open(FH, '<', $ARGV[0]) || die "Can't open $ARGV[0] for read: $!";
+if (defined $ARGV[0] && $ARGV[0] ne '-') {
+    open($fh, '<', $ARGV[0]) or die "$0: $ARGV[0]: $!\n";
+}
+else {
+    $fh = *STDIN;
+}
 
-binmode(FH);
-seek(FH, $offset1, 0) if $offset1;
+binmode $fh;
+if ($offset1 && !seek($fh, $offset1, 0)) {
+    foreach (1 .. $offset1) {
+	$len = read $fh, $data, 1;
```

Comment:
Is there a reason to read here instead of just immediately seek to the offset?

Reply:
* The default behaviour is to seek() if we get a value for -j (L72)
* For a pipe or stdin this will first fail, and the read loop only runs for a failed seek()
* There's no special logic upfront to determine if the file supports seeking